lib/cloudera-manager-client.rb

```diff
@@ -0,0 +1,32 @@
+require 'her'
+require 'cloudera-manager/middleware/url_prefix'
+require 'cloudera-manager/middleware/json_parser'
+require 'cloudera-manager/middleware/json_serializer'
+
+module ClouderaManager
+  def self.api
+    @api ||= Her::API.new
+  end
+
+  def self.setup(options, &blk)
+    username = options.delete(:username)
+    password = options.delete(:password)
+
+    api.setup(options) do |conn|
+      conn.use Her::Middleware::AcceptJSON
+      conn.use ClouderaManager::Middleware::URLPrefix, '/api/v10'
```

Comment:
I guess I liked the prefix being on the outside here such that you could see it, and I copy pasted the url prefix middleware from elsewhere :) I kind of prefer it being obvious like this.